### DISCUSSION TITLE
Replace grub with systemd-boot

### DISCUSSION THREAD

**Discussion by ArKam** [flatcar/Flatcar#1317] [Ideas]
Hi team, I'm currently building my lab using Flatcar on Raspberry PI, It works like a charm from STABLE up to ALPHA channel, as I'm really enjoying flatcar, I'm wondering if it would be relatively easy to replace grub with its systemd-boot counterpart?

Arch linux have a pretty complete and well made documentation about it:
https://wiki.archlinux.org/title/systemd-boot

I know systemd-boot does support A/B Partitioning as systemd-boot works with NixOS, but I'm wondering if there is kind of hidden subtilities that I should know before trying anything?

**Comment by jepio** [comment]
This idea has come up in past discussion, the two issues are:
- systemd-boot does not support legacy BIOS -> we would need to support different bootloaders for UEFI and BIOS
- systemd-boot is not approved for booting with an officially signed secure-boot shim

So it's a neat idea - but has issues.

**Reply by ArKam** [reply]
Nice that someone else asked about it! It means at least I’m not really asking for a cuckoo’s idea, or maybe we’re both cuckoos xD

So regarding the signing for secure boot a way to solve that would somehow to add a « sidecar » service the way archlinux is using pacman hooks to handle that maybe? https://wiki.archlinux.org/title/systemd-boot#Signing_for_Secure_Boot

Regarding BIOS, that why my question initially was rather ask as « Could we have an option to replace that » rather than « Is there a way to make that feature the default ».

Let say that I wouldn’t care about neither BIOS or Secure boot, what should be my better option there?

Loop mount the image and swap everything in there? I doubt this would be that efficient as I guess the update engine will not then be able to find appropriate way to update anymore?

Would the first job rather to make the update engine aware of systemd-boot?

**Reply by pothos** [reply]
I'm playing with the idea of a new image type that uses systemd upstream stuff where possible. Based on the kernel+initrd we use for the PXE case we can build a UKI and put that on a EFI partition with sd-boot. For updating we would define a config for systemd-sysupdate instead of using update-engine.

**Reply by ArKam** [reply]
oooh, that's an interesting idea indeed.
